### DISCUSSION TITLE
Create Shortcut for exe located in sub directory of INSTALLDIR and using HarvestDirectory

### DISCUSSION THREAD

**Discussion by prchdev** [wixtoolset/issues#7890] [Questions]
I am creating MSI using Wix 4.0 command line tools.

I came across a requirement where I want to create shortcut with pointing target exe located in sub folder of INSTALLDIR.
My directory structure is as below:
INSTALLDIR -> C:\Program Files\AppCheck
Folder Structure of AppCheck
-	AppSecure
        o	Somefile.dll
        o	Somefile4.xml
        o	AppBlock
                	Block.exe
-	AppScan
        o	Somefile2.dll
        o	Somefile3.json
        o	AppSet
               	AppSet.exe
I need to create shortcuts for Block.exe and AppSet.exe

I have created Apps.wsx file as below	

```
<?xml version="1.0"?>
<Wix xmlns="http://schemas.microsoft.com/wix/2006/wi">
	<Product Id="*" Name="ApplicationName" Version="1.0.0" Manufacturer="MyCompany" Language="en-US">
	
		<Package InstallerVersion="300" Compressed="yes" Manufacturer="Me" Keywords="Installer,MSI,Database" Comments="This installer database contains the logic and data required to install application" InstallPrivileges="limited"/>
		<Media Id="1" Cabinet="ApplicationName.cab" EmbedCab="yes" CompressionLevel="mszip"/>
		<Directory Id="TARGETDIR" Name="SourceDir">
			
			<Directory Id="ProgramMenuFolder">
			</Directory>
		</Directory>

		<DirectoryRef Id=" ProgramMenuFolder">
			<Component Id="ApplicationShortcut" Guid="PUT-GUID-HERE">
				<Shortcut Id="AppSet" 
                  Name="AppSet"
                  Description="AppSet"
                  Target="[#appset.exe]"
                  WorkingDirectory="TARGETDIR"/>
				<Shortcut Id="Block" 
                  Name="Block"
                  Description="block"
                  Target="[#block.exe]"
                  WorkingDirectory="TARGETDIR"/>
		</Component>
		</DirectoryRef>

		<ItemGroup>
			<HarvestDirectory Include="C://Temp//AppCheck" KeepEmptyDirectories="true" DirectoryRefId=" TARGETDIR " HarvestDirectoryKeepEmptyDirectories="">
				
			</HarvestDirectory>
		</ItemGroup>
		
	</Product>
</Wix>

```
When I am creating msi shortcut getting created with TARGETDIR/INSTALLDIR as working directory instead of subfolder name appscan and appsecure.
Is any option or approach available in Wix 4 to use directory path of subfolder in working directory attribute of shortcut?

Thank you in advance!!!

**Comment by robmen** [comment]
Using heat (aka: the `HarvestDirectory` in your .wixproj) can be tedious. I covered [this topic in the Deployment Dojo episode 52](https://youtube.com/live/XcUmwegYClg). Then, in episode 53, I showed [how better tools make things much easier](https://youtube.com/live/kg6nEBycoFs).
